library/bind.c

```diff
@@ -66,6 +66,7 @@ int ziti_bind(ziti_connection conn, const char *service, const ziti_listen_opts
     conn->type = Server;
     conn->disposer = dispose;
     conn->service = strdup(service);
+    uv_random(NULL, NULL, conn->server.listener_id, sizeof(conn->server.listener_id), 0 , NULL);
```

Comment:
should this be an async call with a timeout, and also some error checking/handling here? [uv_random(http://docs.libuv.org/en/v1.x/misc.html#c.uv_random):
- fails if it can't produce the requested number of random bytes
- may "block indefinitely" (synchronous call), or "never finish" (async call) when there is insufficient entropy.

Reply:
we're only asking for 32 bytes here, it should not be an issue